src/nvim/path.c

```diff
@@ -1993,12 +1993,15 @@ char_u *path_shorten_fname(char_u *full_path, char_u *dir_name)
   size_t len = strlen((char *)dir_name);
   char_u *p = full_path + len;
 
-  if (fnamencmp(dir_name, full_path, len) != 0
-      || !vim_ispathsep(*p)) {
+  if (fnamencmp(dir_name, full_path, len) != 0) {
     return NULL;
   }
 
-  return p + 1;
+  if (vim_ispathsep(*p)) {
+    return p + 1;
+  } else {
+    return p;
+  }
```

Comment:
Maybe a bit shorter, but still readable:

```suggestion
return p + (vim_ispathsep(*p) ? 1 : 0);
```